pkg/controller/eventing/subscription/reconcile.go

```diff
@@ -0,0 +1,331 @@
+	if err != nil {
+		glog.Warningf("Feiled to unserialize legacy target: %s", err)
+		return "", err
+	}
+
+	return t.Status.DomainInternal, nil
+	// Once Knative services support Targetable, switch to using this
+	// 	if t.Status.Targetable != nil {
+	//		return t.Status.Targetable.DomainInternal, nil
+	//	}
+	//return "", fmt.Errorf("status does not contain targetable")
+}
+
+// resolveResult resolves the Spec.Result object.
+func (r *reconciler) resolveResult(namespace string, resultStrategy v1alpha1.ResultStrategy) (string, error) {
+	obj, err := r.fetchObjectReference(namespace, resultStrategy.Target)
+	if err != nil {
+		glog.Warningf("Feiled to fetch ResultStrategy target %+v: %s", resultStrategy, err)
+		return "", err
+	}
+	s := duckv1alpha1.Sink{}
+	err = duck.FromUnstructured(*obj, &s)
+	if err != nil {
+		glog.Warningf("Feiled to unserialize Sinkable target: %s", err)
+		return "", err
+	}
+	if s.Status.Sinkable != nil {
+		return s.Status.Sinkable.DomainInternal, nil
+	}
+	return "", fmt.Errorf("status does not contain targetable")
+}
+
+// resolveFromChannelable fetches an object based on ObjectReference. It assumes that the
+// fetched object then implements Subscribable interface and returns the ObjectReference
+// representing the Channelable interface.
+func (r *reconciler) resolveFromChannelable(namespace string, ref *corev1.ObjectReference) (*duckv1alpha1.Subscription, error) {
+	obj, err := r.fetchObjectReference(namespace, ref)
+	if err != nil {
+		glog.Warningf("Failed to fetch From target %+v: %s", ref, err)
+		return nil, err
+	}
+
+	c := duckv1alpha1.Subscription{}
+	err = duck.FromUnstructured(*obj, &c)
+	return &c, err
+}
+
+// fetchObjectReference fetches an object based on ObjectReference.
+func (r *reconciler) fetchObjectReference(namespace string, ref *corev1.ObjectReference) (*unstructured.Unstructured, error) {
+	//	resourceClient, err := r.CreateResourceInterface2(r.restConfig, ref, namespace)
+	resourceClient, err := r.CreateResourceInterface(namespace, ref)
+	if err != nil {
+		glog.Warningf("failed to create dynamic client resource: %v", err)
+		return nil, err
+	}
+
+	return resourceClient.Get(ref.Name, metav1.GetOptions{})
+}
+
+func (r *reconciler) reconcileFromChannel(namespace string, subscribable corev1.ObjectReference, callDomain string, resultDomain string, deleted bool) error {
+	glog.Infof("Reconciling From Channel: %+v call: %q result %q deleted: %v", subscribable, callDomain, resultDomain, deleted)
+
+	s, err := r.fetchObjectReference(namespace, &subscribable)
+	if err != nil {
+		return err
+	}
+
+	c := duckv1alpha1.Channel{}
+	err = duck.FromUnstructured(*s, &c)
+	if err != nil {
+		return err
+	}
+
+	glog.Infof("Found From Channelable: %+v", c)
+
+	// Add our subscription to the object
+	after := c.DeepCopyObject().(*duckv1alpha1.Channel)
+	after.Spec.Channelable = &duckv1alpha1.Channelable{
+		Subscribers: []duckv1alpha1.ChannelSubscriberSpec{{CallableDomain: "foobar"}},
```

Comment:
I think this should probably use the given `callableDomain` and `resultDomain`

Reply:
haha, yes :) Didn't get a push out before I had to go get to the plane and I lost wifi.